### PR TITLE
Adding React (Enzyme) and Puppeteer Axe-Core Integrations for Accessibility Testing in Spearmint

### PR DESCRIPTION
**Pull request by taoantaoan**
Screenshot from 2021-03-23 11-11-57  

# Adding React (Enzyme) and Puppeteer Axe-Core Integrations for Accessibility Testing in Spearmint                                           

This pull request covers the integration of React (Enzyme) and Puppeteer tests using the axe-core accessibility engine (i.e., npm package) to further augment the Spearmint accessibility testing suite. These two pieces were added as Accessibility Test Case Actions and related files so that a developer can now choose between testing an html file (i.e., Jest), a React component (i.e., Enzyme), or external URL (i.e., Puppeteer).

Axe is an accessibility testing engine for websites and other HTML-based user interfaces. Integrating Enzyme allows users to test React components with axe-core, whereas Puppeteer allows an end-to-end like experience by running the Axe accessibility engine on an external URL of the developer’s choosing. 

## Changes That Took Place:

* Created React component AccTestTypes to handle selection of HTML, React, or Puppeteer type of accessibility test.
* Updated Modal, modalHooks, AccTestMenu, accTestCaseActions, accTestCaseReducer, and other to manage state based on developer selection of type of accessibility test.
* Implemented ability to dynamically generate  Jest, Enzyme, or Puppeteer test code based on developer’s selection.
* Updated directions included in ‘Run Test’ button selection.
* Added types for state items within new React components.
* Update test menu choices to display in alphabetical order.

## Choose Accessibility Test From Main Menu

Screen Shot 2021-03-24 at 1 50 09 PM

Once the developer has chosen ‘Accessibility’ within the Test Menu, he/she/they may choose one of three options from the ‘Choose Type of Accessibility Test’ input: HTML, React (i.e., Enzyme), or Puppeteer. 

This selection will automatically update state, so that upon a selection of the code ‘Preview’ and ‘Run Test’ buttons will show the relevant content based on the type of Accessibility Test selected. See demos below.

# HTML Demo

## HTML Test Code Preview

Screen Shot 2021-03-24 at 1 50 41 PM

## HTML Test Run Directions

Screen Shot 2021-03-24 at 1 50 49 PM

# React Demo

## React Demo Code Preview

Screen Shot 2021-03-24 at 1 51 10 PM

## React Test Run Directions

Screen Shot 2021-03-24 at 1 51 17 PM

<br>

# Puppeteer Demo

## Puppeteer Demo Code Preview

Screen Shot 2021-03-24 at 1 51 58 PM

## Puppeteer Test Run Directions

Screen Shot 2021-03-24 at 1 52 14 PM

<hr>

# Files Created and Modified

```
| src/components/AccTestComponent
    | /AccTestTypes/AccTestTypes.tsx
    | /AccTestTypes/AccTestTypes.module.scss
    | /PuppeteerUrl/PuppeteerUrl.tsx


```
<br>

# Existing Files Modified

```
| src
    | /components
        | /Modals
            | /Modal.jsx
            | /modalHooks.js
        | /SearchInput
            | /SearchInput.scss
        | /TestCase
            | /AccTestCase.tsx
        | /TestMenu
            | /AccTestMenu.tsx
    | /context
        | /actions
            | /accTestCaseActions.ts
        | /reducers
            | /accTestCaseReducer.ts
        | /useGenerateTest.jsx
    | /pages
        | /TestFile/TestFile.jsx
    | /utils
        | /accTypes.ts

```
<br>
<hr>

# Known Bug(s)

<br>
<hr>

# Links to Docs

https://www.deque.com/axe/
https://github.com/dequelabs/axe-core
https://github.com/dequelabs/axe-core/tree/develop/doc/examples
https://www.npmjs.com/package/@axe-core/react
https://www.npmjs.com/package/@axe-core/puppeteer

<hr>

# Contributing Members

Co-authored-by: Annie Shin https://github.com/annieshinn

### CODE REVIEW

File: src/components/TestCase/AccTestCase.tsx [hunk 1]
```diff
@@ -62,20 +63,34 @@ const AccTestCase = () => {
...
       <div id="head">
         <AccTestMenu />
       </div>
 
       <section id={styles.testCaseHeader}>
-        <label htmlFor="fileImport">Import File From</label>
-        <div id={styles.labelInput} style={{ width: '80%' }}>
-          <SearchInput
-            options={Object.keys(filePathMap)}
+        {/* //taotao  */}
```
Comment (annieshinn) [inline src/components/TestCase/AccTestCase.tsx:71]: Just bringing this to your attention! :)

File: src/components/TestMenu/AccTestMenu.tsx [hunk 2]
```diff
@@ -13,7 +13,7 @@ import {
 } from '../../context/actions/globalActions';
 import { AccTestCaseContext } from '../../context/reducers/accTestCaseReducer';
 import { useToggleModal } from './testMenuHooks';
-
+import AccTestTypes from '../AccTestComponent/AccTestTypes/AccTestTypes';
```
Comment (annieshinn) [inline src/components/TestMenu/AccTestMenu.tsx:16]: Is this used?

**Review by annieshinn — APPROVED** [review]
Looks good Tolan! I added a couple comments, but otherwise, all approved!

**Review by bielchristo — APPROVED** [review]

File: src/components/TestCase/AccTestCase.tsx (hunk 1 above)
Comment (taoantaoan) [reply src/components/TestCase/AccTestCase.tsx:71]: Good catch!

File: src/components/TestMenu/AccTestMenu.tsx (hunk 2 above)
Comment (taoantaoan) [reply src/components/TestMenu/AccTestMenu.tsx:16]: Removed. Thanks!

**Review by annieshinn — APPROVED** [review]
Thanks for implementing the changes we discussed. Approved, again! It's raining approvals up in here!

**Pull request merged**